### PR TITLE
[M4] Add M4 v1.4.19

### PR DESCRIPTION
**Pull request by staticfloat**

### CODE REVIEW

File: M/M4/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+script = raw"""
+cd $WORKSPACE/srcdir/m4-*/
+
+./configure --prefix=${prefix} --host=${target}
+make -j${nprocs}
+make install
+"""
+
+# Windows platforms fail with messages about `__memcpy_chk`
```
Comment (giordano) [inline M/M4/build_tarballs.jl:19]: If you really care about Windows, I think you need to link to `libssp` for `__memcpy_chk`

File: M/M4/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+platforms = filter(p -> !Sys.iswindows(p), supported_platforms())
```
Comment (giordano) [inline M/M4/build_tarballs.jl:20]:
```suggestion
platforms = filter(!Sys.iswindows, supported_platforms())
```

File: M/M4/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+products = [
+    ExecutableProduct("m4", :m4),
+]
+
+dependencies = [
```
Comment (giordano) [inline M/M4/build_tarballs.jl:26]:
```suggestion
dependencies = Dependency[
```

File: M/M4/build_tarballs.jl (hunk 1 above)
Comment (staticfloat) [reply M/M4/build_tarballs.jl]: I don't care about Windows (this is for host tools in BB2) but looks like `-fstack-protector` is what is required to get this to work; thanks for the tip!

**Pull request merged**